### PR TITLE
IGNITE-12969 TxRecovery discovery listener should implement HighPriorityListener	

### PR DESCRIPTION
**Pull request by anton-vinogradov**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/NearTxResultHandler.java [hunk 1]
```diff
@@ -108,6 +109,9 @@ public static GridNearTxEnlistResponse createResponse(GridDhtTxEnlistFuture fut)
         try {
             cctx.io().send(nearNodeId, res, cctx.ioPolicy());
         }
+        catch (ClusterTopologyCheckedException e) {
```
Comment (nizhikov) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/NearTxResultHandler.java:112]: Why this change?

Comment (anton-vinogradov) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/NearTxResultHandler.java:112]: Code speed-up increase probability of this case. 
It was a problem before, but very rarely.
